source/common/tcp_proxy/factory.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include "common/tcp_proxy/upstream_interface.h"
+
+namespace Envoy {
+
+namespace Upstream {
+class LoadBalancerContext;
+class ClusterManager;
+} // namespace Upstream
+
+namespace TcpProxy {
+
+// TODO(lambdai): move to include path.
+class TcpUpstreamFactory {
+public:
+  virtual ~TcpUpstreamFactory() = default;
+  virtual ConnectionHandlePtr createTcpUpstreamHandle(
```

Comment:
I think we're going to want the TCP factory to look like the HTTP factory, where we actually have a factory which creates the GenericConnPoolPtr, then from that we can create the upstream
It'd be good to have similar naming and, as much as possible, similar APIs

Reply:
I had a bad understanding of life time at GenericConnPool. Close looking into the http factory, the major change in TCP is naming. Working on it...